e2e/support/helpers/component-testing-sdk/component-embedding-sdk-question-helpers.tsx

```diff
@@ -0,0 +1,49 @@
+import {
+  InteractiveQuestion,
+  StaticQuestion,
+} from "@metabase/embedding-sdk-react";
+import type { ComponentProps } from "react";
+
+import { mountSdkContent } from "./component-embedding-sdk-helpers";
+
+interface Options {
+  shouldAssertCardQuery?: boolean;
+}
+
+export function mountInteractiveQuestion(
+  extraProps: Partial<ComponentProps<typeof InteractiveQuestion>> = {},
+  { shouldAssertCardQuery }: Options = { shouldAssertCardQuery: true },
+) {
+  cy.intercept("GET", "/api/card/*").as("getCard");
+  cy.intercept("POST", "/api/card/*/query").as("cardQuery");
+
+  cy.get<number>("@questionId").then(questionId => {
+    mountSdkContent(
+      <InteractiveQuestion questionId={questionId} {...extraProps} />,
+    );
+  });
+
+  if (shouldAssertCardQuery) {
```

Comment:
This is added, previously we only assert questions that will load, but now we will want to test questions that would fail to load as well.

Reply:
I wonder if it makes sense to assert a 404 response as well for questions that failed to load?

Reply:
not in all cases that we'd call the card/dashboard endpoint. If we couldn't resolve the entity ID we'd not call such endpoints, so `wait("@getCard")` or whatever would fail.

This is from removing the if check from one test.
![image](https://github.com/user-attachments/assets/f2b3a96c-8f7c-47aa-ad37-8ea858c51274)